File: Java/src/covariance/informatics/CovNode.java
package covariance.informatics;

import covariance.abs.AbsGraph;
import covariance.abs.AbsVertex;

public class CovNode extends AbsVertex
{
    public CovNode(Integer label)
    {
        super(label);
    }
    
    public void setGraph(CovNetwork g)
    {
        super.setGraph(g);        
    }
    
    @Override
    public void setGraph(AbsGraph g)
    {
        try
        {
            setGraph((CovNetwork) g);
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
    }
}

File: scala/src/covariance/informatics/CovNode.scala
package covariance.informatics

import covariance.abs.AbsGraph
import covariance.abs.AbsVertex

class CovNode(label: Integer) extends AbsVertex(label) {

  def setGraph(g: CovNetwork): Unit = {
    super.setGraph(g)
  }

  override def setGraph(g: AbsGraph): Unit = {
    try {
      setGraph(g.asInstanceOf[CovNetwork])
    } catch {
      case e: Exception =>
        e.printStackTrace()

    }
  }

}
